zookeeper-server/src/test/java/org/apache/zookeeper/test/SaslSuperUserTest.java

```diff
@@ -127,11 +128,35 @@ public void testSuperIsSuper() throws Exception {
             zk.delete("/digest_read", -1);
             zk.delete("/sasl_read/sub", -1);
             zk.delete("/sasl_read", -1);
-            //If the test failes it will most likely fail with a NoAuth exception before it ever gets to this assertion
-            assertEquals(authFailed.get(), 0);
         } finally {
             zk.close();
         }
     }
 
+    @Test
+    public void testSuperIsSuper() throws Exception {
+        connectAndPerformSuperOps();
+        //If the test fails it will most likely fail with a NoAuth exception before it ever gets to this assertion
+        assertEquals(authFailed.get(), 0);
+    }
+
+    @Test
+    public void testOtherSuperIsSuper() throws Exception {
+        String prevSection = System.setProperty(ZKClientConfig.LOGIN_CONTEXT_NAME_KEY, "OtherClient");
+
+        // KLUDGE: We do this quite late, as the server has been
+        // started at this point--but current the implementation looks
```

Comment:
typo: "but current the implementation looks"
-> "but the implementation currently looks"

Reply:
Fixed; thanks!

Reply:
(I'll merge this if I get a green CI.)